Add initial community analyzers

srijan-deepsource opened the pull request:
Ref: ENG-36

srijan-deepsource committed:
chore: add initial structure

srijan-deepsource committed:
chore: add more required stuff

srijan-deepsource committed:
chore: Add more structures

Still playing around. Nothing is final. Yet.

srijan-deepsource committed:
feat: Add issues and utilities for kubelinter

srijan-deepsource committed:
Update structure

Put everything related to deepsource under `.deepsource` sub dir.

srijan-deepsource committed:
Add cloudbuild

srijan-deepsource committed:
add worklow to emit sync signal

srijan-deepsource committed:
Fix logo

srijan-deepsource committed:
fix version

srijan-deepsource committed:
Convert to svg

srijan-deepsource commented on cloudbuild.yaml:
```diff
@@ -0,0 +1,18 @@
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
+        'https://deepsource.io/internal/graphql/'
```
```suggestion
        'https://app.deepsource.one/internal/graphql/'
```

srijan-deepsource commented on cloudbuild.yaml:
```diff
@@ -0,0 +1,18 @@
+        '--request', 'POST',
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
```
```suggestion
        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME","analyzer_version_map":{\"infer\":\"v1.1.0\",\"kube-linter\":\"0.6.4\"}}}}',
```

srijan-deepsource commented on cloudbuild_dev.yaml:
```diff
@@ -0,0 +1,18 @@
+        '--request', 'POST',
+        '--header', "Authorization: Bearer ${_DATASYNC_SECRET}",
+        '--header', 'Content-Type: application/json',
+        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME", "analyzer_version_map": {"infer": "v1.1.0", "kube-linter": "0.6.4"}}}}',
```
```suggestion
        '--data-raw', '{"query":" mutation($input: TriggerAnalyzerSyncInput!){\n    triggerAnalyzerSync(input: $input){\n        ok\n    }\n}","variables":{"input":{"repoName":"community-analyzers","tag":"$TAG_NAME","analyzer_version_map": {\"infer\":\"v1.1.0\",\"kube-linter\":\"0.6.4\"}}}}',
```

srijan-deepsource committed:
Apply suggestions from code review

Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

srijan-deepsource commented on analyzers/kube-linter/.deepsource/analyzer/analyzer.toml:
```diff
@@ -0,0 +1,8 @@
+name = "KubeLinter"
+shortcode = "kube-linter"
+status = "active"
+processors = ["skipcq", "source_code_load"]
```
```suggestion
processors = ["skip_cq", "source_code_load"]
```

srijan-deepsource replied:
```suggestion
```

srijan-deepsource committed:
Update analyzers/kube-linter/.deepsource/analyzer/analyzer.toml

Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

tushar-deepsource commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
```
```suggestion
*$py.class
venv*
.tox
.coverage
```

tushar-deepsource commented on analyzers/kube-linter/utils/issue_map.json:
```diff
@@ -0,0 +1,155 @@
+    "writable-host-mount": {
+        "issue_code": "KUBELIN-W1051"
+    }
+}
```
```suggestion
}

```

tushar-deepsource commented on analyzers/infer/utils/issue_scraper.py:
```diff
@@ -0,0 +1,36 @@
+                pass
+            infer_issue_id = filename.removesuffix(".md")
+
+        breakpoint()
```
?

srijan-deepsource replied:
oops

srijan-deepsource committed:
Apply suggestions

tushar-deepsource commented on analyzers/infer/utils/issue_scraper.py:
```diff
@@ -0,0 +1,34 @@
+    clone_command = f"git clone {CLONE_URL} --depth=1".split()
+    with TemporaryDirectory() as temp_dir_name:
+        subprocess.run(clone_command, cwd=temp_dir_name)
+        resp = subprocess.run(["ls", temp_dir_name])
```
```suggestion
```

tushar-deepsource committed:
Update analyzers/infer/utils/issue_scraper.py

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

deepsource-autofix[bot] committed:
style: format code with Black

This commit fixes the style issues introduced in 65fb8bf according to the output
from Black.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/1

tushar-deepsource commented on analyzers/kube-linter/utils/issue_map_gen.py:
```diff
@@ -0,0 +1,53 @@
+def get_next_code(mapping) -> str:
+    """Return the next available issue code."""
+    num_issues = len(mapping.keys())  # get the number of issues already in the mapping
+    next_code = 1000 + num_issues + 1  # issue code series starts from `1001`
```
```suggestion
    next_code = 1001 + num_issues  # issue code series starts from `1001`
```

tushar-deepsource committed:
Update analyzers/kube-linter/utils/issue_map_gen.py

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource commented on analyzers/kube-linter/utils/issue_map_gen.py:
```diff
@@ -0,0 +1,53 @@
+    next_code = 1001 + num_issues  # issue code series starts from `1001`
+    while True:
+        yield next_code
+        next_code += 1
```
```suggestion
    yield from itertools.count(next_code)
```

tushar-deepsource commented on analyzers/kube-linter/utils/issue_map_gen.py:
```diff
@@ -0,0 +1,53 @@
+import subprocess
+import os
+import json
```
```suggestion
import itertools
import json
import os
import subprocess
```

tushar-deepsource committed:
Use itertools.count

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

tushar-deepsource commented on analyzers/infer/utils/issue_scraper.py:
```diff
@@ -0,0 +1,34 @@
+        infer_issues_dir = f"{temp_dir_name}/infer/infer/documentation/issues"
+        for filename in os.listdir(infer_issues_dir):
+            if not filename.endswith(".md"):
+                pass
```
```suggestion
                continue
```

tushar-deepsource committed:
pass -> continue

Signed-off-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>

srijan-deepsource committed:
Merge branch 'master' into initial-structure

Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in 7ea78a6 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/1

deepsource-autofix[bot] committed:
refactor: autofix issues in 2 files

Resolved issues in the following files with DeepSource Autofix:
1. analyzers/infer/utils/issue_scraper.py
2. analyzers/kube-linter/utils/issue_map_gen.py

tushar-deepsource reviewed (approved):
LGTM
